### DISCUSSION TITLE
Not understanding of notations in paper: Direct Position-Based Solver for Stiff Rods

### DISCUSSION THREAD

**Discussion by ou-ais** [InteractiveComputerGraphics/PositionBasedDynamics#160] [Q&A]
Thank you for your contribution about position-based simulation of flexible rod.

I am reading the paper "Direct Position-Based Solver for Stiff Rods" and try to implement it by myself. However, there is one equation that I can not understand. I would appreciate it if you could tell me.

The meaning of **p** and **x** appearing in the first three rows of Eq. (23) is not understood. Actually, I do not know the difference between **p** and **x**. 

![image](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/assets/49639711/b509fbb5-520b-4847-84ce-d59ba23d7446)

For **p**, from the following description of Eq. (23),  **p**1 and **p**2 is a point on segment 1 and segment 2, respectively, that are given in the local coordinates. Futher, as described in P.5 2nd paragraph, **p** denotes the position of each segments's center of mass. According to these interpretations, **p** will be the origin of the local coordinate, i.e. **p**=(0, 0, 0). Then, the first three rows of Eq. (23) will be **x1**-**x2**, is that so?

For **x**, my understanding is that **x** is the current position of the center of each segment in world coordinate during Newton iterations, am I right?

Expect your reply.

**Comment by janbender** [comment]
Sorry for the late reply. x denotes the center of an segment. R is the rotation around the center and p is a connector point on the segment.
